File: Problems/No.14_Longest_Common_Prefix/longest_common_prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        int strs_length = strs.size();  // the length of strs
        string result="";               // save the result
        sort(strs.begin(), strs.end()); // sort strs by alphabet order
        
        // start compute
        for (int i=0; i<strs[0].length(); i++) {
            
            // check whether the alphabet is all same in index i
            if (strs[0][i] == strs[strs_length-1][i]) result += strs[0][i];
            else break;
        }
        
        return result;
    }
};

File: Problems/No.14_Longest_Common_Prefix/longest_common_prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        
        result = ""     # save the result
        strs.sort()     # sort strs in alphabet order
        
        # start compute
        for i in range(len(strs[0])):
            
            # check whether the alphabet is all same in index i
            if (strs[0][i] == strs[-1][i]):
                result = result + strs[0][i]
            else:
                break
        
        return result
        
